app/post.go

```diff
@@ -28,14 +30,29 @@ func mdToHTML(md []byte) []byte {
 }
 
 func postHandler(c *gin.Context, app_settings common.AppSettings, database database.Database) ([]byte, error) {
+
+	// Get the post ID from the URL
+	postIDStr := c.Param("id")
+
+	// converting our postID string to an integer
+	postID, err := strconv.Atoi(postIDStr)
+
+	if err != nil || postID < 0 {
```

Comment:
Looks better now, but you should update the test `TestPostFailureStringKey` so we expect `http.StatusBadRequest` instead of the `http.StatusInternalServerError` - The tests should pass after that.